core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -399,6 +399,8 @@ TranslogWriter createWriter(long fileGeneration) throws IOException {
         return newFile;
     }
 
+    final AtomicBoolean foldingGeneration = new AtomicBoolean();
```

Comment:
Putting this down here rather than up with the rest of the members will make it much harder to find. Even if this is the one place it is used.